openstates/md/bills.py

```diff
@@ -165,10 +170,132 @@ def parse_bill_votes(self, doc, bill):
                 vote.add_source(href)
                 yield vote
 
+    def parse_bill_votes_new(self, doc, bill):
+        elems = doc.xpath('//a')
+
+        # MD has a habit of listing votes twice
+        seen_votes = set()
+
+        for elem in elems:
+            href = elem.get('href')
+            if (href and "votes" in href and href.endswith('pdf') and
+                    ("Senate" in href or "House" in href) and href not in seen_votes):
+                seen_votes.add(href)
+                vote = self.parse_vote_pdf(href, bill)
+                vote.add_source(href)
+                yield vote
+
+    def parse_vote_pdf(self, vote_url, bill):
+
+        filename, response = self.urlretrieve(vote_url)
+
+        text = convert_pdf(filename, type='text').decode()
+        lines = text.splitlines()
+
+        if 'senate' in vote_url:
+            chamber = 'upper'
+        else:
+            chamber = 'lower'
+
+        date_string = lines[0].split('Date:')[1].strip()
+        date = datetime.datetime.strptime(date_string, "%b %d, %Y %I:%M (%p)")
+
+        pageIndex = -1
+        for index, line in enumerate(lines):
+            if 'Yeas' in line and 'Nays' in line:
+                pageIndex = index
+                break
+
+        yes_count = -1
+        no_count = -1
+        other_count = 0
+
+        if pageIndex > 0:
+
+            counts = re.split(r'\s{2,}', lines[pageIndex].strip())
+
+            for count in counts:
+                number, string = count.split(' ', 1)
+                number = int(number)
+
+                if string == 'Yeas':
+                    yes_count = number
+                elif string == 'Nays':
+                    no_count = number
+                else:
+                    other_count += number
+
+        passed = yes_count > no_count
+
+        motion = re.split(r'\s{2,}', lines[pageIndex-3].strip())[0]
+        if motion == '':
+            motion = "No motion given"
```

Comment:
Add a warning here, at least. I might even be inclined to `assert motion`. Are there any cases where a motion isn't found, for the current session?

Reply:
+1